lib/eksekuter.rb

```diff
@@ -14,71 +14,60 @@ def initialize logger: nil
     @stdin_buffer = nil
   end
 
-  def run *args, **opts, &block
-    env, cmd = separate_env_and_cmd(args)
-    # p opts
-    capture = opts.delete(:capture) || false
-    streams = get_out_err_streams(capture)
+  alias kernel_spawn spawn
 
-    set_streams_in_opts streams, opts
-
-    params = { env: env, cmd: cmd, opts: opts, block: block }
+  # Wraps around Kernel#spawn so that the return value is a EksekResult.
+  # @return EksekResult
+  def spawn *args, **opts
+    env, cmd = split_env_and_cmd(args)
+    params = { env: env, cmd: cmd, opts: opts }
+    process_status = spawn_and_get_status(params)
+    assemble_result(params, process_status)
+  end
 
-    process_status = run_process(params, streams)
-    assemble_result(params, process_status, streams)
+  # Like Eksekuter#spawn but the :out and :err options are ignored; instead
+  # the output is attached to the EksekResult object returned.
+  # Returns a EksekResult object which getters :stdout and :stderr containing
+  # the corresponding output of the spawned process.
+  # @return EksekResult
+  def capture *args, **opts
+    env, cmd = split_env_and_cmd(args)
+    out_read, out_write = IO.pipe
+    err_read, err_write = IO.pipe
+    opts2 = opts.merge(out: out_write, err: err_write)
+    params = { env: env, cmd: cmd, opts: opts2 }
+    process_status = spawn_and_get_status(params)
+    out_write.close
+    err_write.close
+    assemble_result(params, process_status, out_read, err_read)
   end
 
   private
 
-  def separate_env_and_cmd args
-    env = args[0].is_a?(Hash) ? args.shift : {}
+  def split_env_and_cmd args
+    args2 = args.clone
```

Comment:
Can we not name this `args2` but maybe `cmd_args` or something?

Reply:
Done :)